.github/workflows/build-fd-image.yaml

```diff
@@ -18,76 +18,73 @@ permissions:
   packages: write
 
 jobs:
-  build-amd64-prod-image-metadata:
+  determine-tags:
```

Comment:
Instead of "determining" the tabs in each "metadata" step -- we just do it once at the top of the workflow and then reference the outputs (tags) in other steps.  This ensures that the tags are propagated correctly when copying the images from GHCR to Dockerhub.